### PR TITLE
Bugfix: Re-applying of featuresets does not update label changes

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:
Currently, when we reapply featuresets to add/remove labels from featureset or features, the changes made to labels are not propagated.
**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: core/src/main/java/feast/core/model/Feature.java [hunk 1]
```diff
@@ -235,11 +235,12 @@ public void archive() {
   }
 
   /**
-   * Update the feature object with a valid feature spec. Only schema changes are allowed.
+   * Update the feature object with a valid feature spec.
    *
    * @param featureSpec {@link FeatureSpec} containing schema changes.
    */
   public void updateFromProto(FeatureSpec featureSpec) {
+    this.setLabels(TypeConversion.convertMapToJsonString(featureSpec.getLabelsMap()));
```
Comment (pyalex) [inline core/src/main/java/feast/core/model/Feature.java:243]: shouldn't we apply update after isArchived check?

File: core/src/main/java/feast/core/model/FeatureSet.java [hunk 2]
```diff
@@ -314,9 +314,10 @@ public void updateFromProto(FeatureSetProto.FeatureSet featureSetProto)
               spec.getEntitiesList(), existingEntities));
     }
 
-    // 4. Update max age and source.
+    // 4. Update max age, source and labels.
     maxAgeSeconds = spec.getMaxAge().getSeconds();
```
Comment (pyalex) [inline core/src/main/java/feast/core/model/FeatureSet.java:318]: could please add `this.` here as well

File: core/src/main/java/feast/core/model/FeatureSet.java [hunk 3]
```diff
@@ -314,9 +314,10 @@ public void updateFromProto(FeatureSetProto.FeatureSet featureSetProto)
               spec.getEntitiesList(), existingEntities));
     }
 
-    // 4. Update max age and source.
+    // 4. Update max age, source and labels.
```
Comment (pyalex) [inline core/src/main/java/feast/core/model/FeatureSet.java:317]: looks like numbering is broken. should be `2.`

File: core/src/main/java/feast/core/model/FeatureSet.java (hunk 2 above)
Comment (terryyylim) [reply core/src/main/java/feast/core/model/FeatureSet.java:318]: adding for source as well.

**Pull request merged**